packages/runtime/plugin-runtime/src/ssr/serverRender/renderToString/entry.ts

```diff
@@ -93,6 +103,8 @@ export default class Entry {
     this.App = options.App;
     this.pluginConfig = config;
 
+    this.reporter = createSSRReporter(ctx.reporter);
```

Comment:
where the `ctx.reporter ` inject..I can't find it

Reply:
You can see those [codes](https://github.com/web-infra-dev/modern.js/pull/4333/files#diff-e52c9591859027aaebe506a842d26a586fc00cbe4ac3acecdd26ac608d557465R40-R45).

we inject a defaultReporter that no implement any methods.

Reply:
Then any server plugin can override the default-reporter by `beforeProdServer` Hook